numpy/core/src/private/npy_fpmath.h

```diff
@@ -7,39 +7,10 @@
 #include "numpy/npy_cpu.h"
 #include "numpy/npy_common.h"
 
-#ifdef NPY_OS_DARWIN
-    /* This hardcoded logic is fragile, but universal builds makes it
-     * difficult to detect arch-specific features */
-
-    /* MAC OS X < 10.4 and gcc < 4 does not support proper long double, and
-     * is the same as double on those platforms */
-    #if NPY_BITSOF_LONGDOUBLE == NPY_BITSOF_DOUBLE
-        /* This assumes that FPU and ALU have the same endianness */
-        #if NPY_BYTE_ORDER == NPY_LITTLE_ENDIAN
-            #define HAVE_LDOUBLE_IEEE_DOUBLE_LE
-        #elif NPY_BYTE_ORDER == NPY_BIG_ENDIAN
-            #define HAVE_LDOUBLE_IEEE_DOUBLE_BE
-        #else
-            #error Endianness undefined ?
-        #endif
-    #else
-        #if defined(NPY_CPU_X86)
-            #define HAVE_LDOUBLE_INTEL_EXTENDED_12_BYTES_LE
-        #elif defined(NPY_CPU_AMD64)
-            #define HAVE_LDOUBLE_INTEL_EXTENDED_16_BYTES_LE
-        #elif defined(NPY_CPU_PPC) || defined(NPY_CPU_PPC64)
-            #define HAVE_LDOUBLE_IEEE_DOUBLE_16_BYTES_BE
-        #elif defined(NPY_CPU_PPC64LE)
-            #define HAVE_LDOUBLE_IEEE_DOUBLE_16_BYTES_LE
-        #endif
-    #endif
-#endif
-
 #if !(defined(HAVE_LDOUBLE_IEEE_QUAD_BE) || \
       defined(HAVE_LDOUBLE_IEEE_QUAD_LE) || \
       defined(HAVE_LDOUBLE_IEEE_DOUBLE_LE) || \
       defined(HAVE_LDOUBLE_IEEE_DOUBLE_BE) || \
-      defined(HAVE_LDOUBLE_IEEE_DOUBLE_16_BYTES_BE) || \
```

Comment:
Why did this disappear?

Reply:
So I just refreshed myself on the thinking here.

First, with the deletions above and in `npy_math_private.h` this variable is no longer used anywhere and we should error if we see it. Also, note that `HAVE_LDOUBLE_IEEE_DOUBLE_16_BYTES_LE` should have been here too but wasn't, but is also no longer used anywhere.

As to the deeper question of the deletion of all this darwin stuff: By my reading, there is some confusion in the existing numpy code about the floating-point implementation of (apple) ppc systems. In fact, long double on ppc systems is a "ibm double-double" value, which numpy detects under the name `HAVE_LDOUBLE_DOUBLE_DOUBLE_*`. Therefore the name `HAVE_LDOUBLE_IEEE_DOUBLE_16_BYTES_BE` is superflous/duplicate, and can be replaced by `HAVE_LDOUBLE_DOUBLE_DOUBLE_BE`.

See asterisk here:
https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/LowLevelABI/100-32-bit_PowerPC_Function_Calling_Conventions/32bitPowerPC.html

Furthermore, I think the treatment of  `HAVE_LDOUBLE_IEEE_DOUBLE_16_BYTES_LE` in `npy_math_private.h` is incorrect: It treats the first 8 bytes as a normal double and then completely ignores the second 8 bytes. This doesn't cause numpy to blow up because the first double of a double-double type gives the "right" value and by ignoring the second double we just lost some precision. But, the fact that we were losing precision is bad.

Lastly, now that this confusion is cleared up, it seems to me we no longer need the darwin specific ifdef above, we can let the code in `setup_common.py` detect the double-double type like on other systems.

Reply:
> Lastly, now that this confusion is cleared up, it seems to me we no longer need the darwin specific ifdef above, we can let the code in setup_common.py detect the double-double type like on other systems.

Can you make a standalone PR for just this?

Reply:
sure, coming up

Reply:
See #11191